blog/15-02-2023-how-to-tokenize-physical-assets-on-celo/how-to-tokenize-physical-assets-on-celo.md

```diff
@@ -0,0 +1,300 @@
+    
+*6.  import "@openzeppelin/contracts/access/Ownable.sol";:* This is an import statement that brings in the Ownable contract from the OpenZeppelin library. This contract provides the functionality to set an owner for the token contract.
+    
+*7.  contract TokenizationOfPhysicalAsset is ERC20, ERC20Burnable, Pausable, Ownable {:* This is the start of the main contract, which is named TokenizationOfPhysicalAsset. It inherits from the ERC20, ERC20Burnable, Pausable, and Ownable contracts that were imported earlier.
+    
+*8.  constructor() ERC20("Tokenization of Physical Asset", "Tokenize") {:* This is the constructor for the contract, which is called when the contract is first deployed. It initializes the token with a name of "Tokenization of Physical Asset" and a symbol of "Tokenize". The ERC20 contract's constructor is also called, which sets up some initial variables.
+    
+*9.  _mint(msg.sender, 100 * 10 ** decimals());:* This function call mints 100 tokens to the address that deployed the contract, which is represented by msg.sender. The number is multiplied by 10 ** decimals(), which sets the number of decimal places for the token. By default, the ERC20 standard has 18 decimal places, so this call will mint 100 tokens with 18 decimal places.
+    
+*10.  function pause() public onlyOwner {:* This function is used to pause the token contract. It can only be called by the contract owner.
+    
+*11.  _pause();:* This is the function call that actually pauses the token contract. It is provided by the Pausable contract.
+    
+*12.  function unpause() public onlyOwner {:* This function is used to unpause the token contract. It can only be called by the contract owner.
+    
+*13.  _unpause();:* This is the function call that actually unpauses the token contract. It is provided by the Pausable contract.
+    
+*14.  function mint(address to, uint256 amount) public onlyOwner {:* This function is used to mint new tokens to a specific address. It can only be called by the contract owner.
+    
+*15.  function _beforeTokenTransfer(address from, address to, uint256 amount) internal whenNotPaused override {* This is a function that is defined in the TokenizationOfPhysicalAsset contract. It overrides the _beforeTokenTransfer function inherited from the ERC20 contract, which is called before every transfer of tokens.
+    
+  
+
+**ERC-721** is a standard for creating non-fungible tokens, which are unique tokens that represent one-of-a-kind assets. ERC-721 tokens are also created using smart contracts and are stored in Ethereum wallets. They can be traded on exchanges and used to represent ownership of unique assets.
+
+To tokenize physical assets on Celo, you can use a standard called ERC-721.You can also start your contract with Openzeppelin and then customize for yours needs.
```

Comment:
To tokenize physical assets on Celo, you can use a standard called ERC-721. You can also start your contract with Openzeppelin and then customize it for your needs.

Reply:
Done